### PR TITLE
Added solved task-16-multithreading(advanced)

### PR DESCRIPTION
**Pull request by SerivljaEvgen**

### CODE REVIEW

File: src/main/java/homework/task_16_multithreading_advanced/PetrolStation.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package homework.task_16_multithreading_advanced;
+
+public class PetrolStation {
+
+    private int amount; //кількість пального на станції.
+    private ThreadSafeList<Integer> refuelRequest = new ThreadSafeList<>();
```
Comment (kartius) [inline src/main/java/homework/task_16_multithreading_advanced/PetrolStation.java]: Better to use AtomicInteger here. You don't need to use a list

File: src/main/java/homework/task_16_multithreading_advanced/Main.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+public class Main {
+    public static void main(String[] args) {
+
+        PetrolStation petrolStation = new PetrolStation(100);
+
+        Thread car1 = new Thread(()-> petrolStation.doRefuel(30));
+
+        Thread car2 = new Thread(()-> petrolStation.doRefuel(20));
+
```
Comment (kartius) [inline src/main/java/homework/task_16_multithreading_advanced/Main.java:13]: Please, use ExecutorService with predefined pool size and use submit to start thread logic

File: src/main/java/homework/task_16_multithreading_advanced/PetrolStation.java (hunk 1 above)
Comment (SerivljaEvgen) [reply src/main/java/homework/task_16_multithreading_advanced/PetrolStation.java]: corrected

File: src/main/java/homework/task_16_multithreading_advanced/Main.java (hunk 2 above)
Comment (SerivljaEvgen) [reply src/main/java/homework/task_16_multithreading_advanced/Main.java:13]: corrected

File: src/main/java/homework/task_16_multithreading_advanced/Main.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        PetrolStation petrolStation = new PetrolStation(100, maxRequests);
+
+
+        ExecutorService executor = Executors.newFixedThreadPool(maxRequests); // Створюємо пул потоків з розміром 4.
+
+        // Використовуємо ExecutorService для запуску потоків замість створення окремих потоків.
+        executor.submit(() -> petrolStation.doRefuel(30));
+        executor.submit(() -> petrolStation.doRefuel(20));
+        executor.submit(() -> petrolStation.doRefuel(15));
+        executor.submit(() -> petrolStation.doRefuel(10));
```
Comment (kartius) [inline src/main/java/homework/task_16_multithreading_advanced/Main.java:18]: Better to use foreach to iterate submit operation with random amount of fuel

**Review by kartius — APPROVED** [review]

**Pull request merged**